internal/contracts/manager.go

```diff
@@ -349,7 +349,12 @@ func (cm *contractManager) DeployContract(ctx context.Context, req *core.Contrac
 }
 
 func (cm *contractManager) InvokeContract(ctx context.Context, req *core.ContractCallRequest, waitConfirm bool) (res interface{}, err error) {
-	req.Key, err = cm.identity.ResolveInputSigningKey(ctx, req.Key, identity.KeyNormalizationBlockchainPlugin)
+	keyResolver := cm.identity.ResolveInputSigningKey
```

Comment:
Note I went for separate methods on the identity interface (even though it makes this calling code more complex), so as to very clearly distinguish the query case and try to minimize the risk of any accidental future use of the `query` intent on other code paths.